File: algorithms/1_easy/C/03870_count_commas_in_range/solution.c
int countCommas(const int n_int) {
    if (n_int < 1000) { return 0; }

    long long n = n_int;
    long long base = 999;
    long long cnt = 0;

    while (base <= n) {
        cnt += (n - base);
        base = base * 1000 + 999;
    }

    return (int)cnt;
}

File: algorithms/1_easy/C/03870_count_commas_in_range/solution.rs
impl Solution {
    pub fn count_commas(n: i32) -> i32 {
        if n < 1000 { return 0; }

        let n = n as i64;
        let mut base: i64 = 999;
        let mut cnt: i64 = 0;

        while base <= n {
            cnt += (n - base);
            base = base * 1000 + 999;
        }

        cnt.try_into().unwrap()
    }
}
